src/viam/sdk/common/exception.hpp

```diff
@@ -0,0 +1,66 @@
+/// @file common/exception.hpp
+///
+/// @brief Defines custom exceptions for the SDK.
+#pragma once
+#include <grpcpp/support/status.h>
+#include <stdexcept>
+#include <string>
+
+#include <viam/sdk/resource/resource_api.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup Exception Classes related to SDK exceptions.
+
+/// @class ErrorCondition
+/// @brief Defines a set of a error conditions to be used in conjunction with
+/// Exception.
+/// @ingroup Exception
+enum class ErrorCondition : uint8_t {
+    k_general = 0,             // Default condition
+    k_connection,              // Issue during connection establishment
```

Comment:
Let's give all of these explicit values for API stability reasons. Yes, they will get them by default, but if someone later (unwisely) added one in the middle, bad things would result.

Reply:
Cool; done.